Code/Sandbox/Editor/CryEdit.cpp

```diff
@@ -1510,34 +1510,97 @@ void CCryEditApp::RunInitPythonScript(CEditCommandLineInfo& cmdInfo)
+                for (AZStd::string_view file : fileList)
                 {
-                    pythonTestCase = cmdInfo.m_pythontTestCase.toUtf8().constData();
+                    testList.push_back({ file, {} });
                 }
 
-                EditorPythonRunnerRequestBus::Broadcast(&EditorPythonRunnerRequestBus::Events::ExecuteByFilenameAsTest, cmdInfo.m_strFileName.toUtf8().constData(), pythonTestCase, pythonArgs);
+                QByteArray testCaseStr = cmdInfo.m_pythontTestCase.toUtf8();
+                FileAndTestCase* testListIt = testList.begin();
+                AzFramework::StringFunc::TokenizeVisitor(testCaseStr.constData(),
+                    [&testList, &testListIt](AZStd::string_view elem)
+                    {
+                        if (testListIt != testList.end())
+                        {
+                            testListIt->m_testCase = elem;
+                            ++testListIt;
+                        }
+                    }, ';', false /* keepEmptyStrings */
+                );
 
-                // Close the editor gracefully as the test has completed
-                GetIEditor()->GetDocument()->SetModifiedFlag(false);
-                QTimer::singleShot(0, qApp, &QApplication::closeAllWindows);
+                bool success = true;
+                AZStd::for_each(testList.begin(), testList.end(),
+                    [&success, &pythonArgs](const FileAndTestCase& test)
+                    {
+                        bool fileSucceded = false;
+                        EditorPythonRunnerRequestBus::BroadcastResult(fileSucceded,
+                            &EditorPythonRunnerRequestBus::Events::ExecuteByFilenameAsTest, test.m_file, test.m_testCase, pythonArgs);
+                        success = success && fileSucceded;
+                    }
+                );
+
+                if (success)
+                {
+                    // Close the editor gracefully as the test has completed
+                    GetIEditor()->GetDocument()->SetModifiedFlag(false);
+                    QTimer::singleShot(0, qApp, &QApplication::closeAllWindows);
+                }
+                else
+                {
+                    // Close down the application with 0xF exit code indicating failure of the test
+                    AZ::Debug::Trace::Terminate(0xF);
+                }
             }
             else
             {
-                EditorPythonRunnerRequestBus::Broadcast(&EditorPythonRunnerRequestBus::Events::ExecuteByFilenameWithArgs, cmdInfo.m_strFileName.toUtf8().constData(), pythonArgs);
+                AZStd::for_each(fileList.begin(), fileList.end(),
+                    [&pythonArgs](AZStd::string_view filename)
+                    {
+                        EditorPythonRunnerRequestBus::Broadcast(
+                            &EditorPythonRunnerRequestBus::Events::ExecuteByFilenameWithArgs, filename, pythonArgs);
+                    }
+                );
```

Comment:
It actually more efficient to pass in a lambda that iterates over the fileList to the `Broadcast` call directly
This avoids having to invoke the EBus multiple times.

```suggestion
                auto ExecuteFilenamesWithArgs = [&pythonArgs, &fileList](EditorPythonRunnerRequests* pythonRunnerRequests)
                {
                    for (AZStd::string_view filename : fileList)
                    {
                        pythonRunnerRequests->ExecuteByFilenameWithArgs(filename, pythonArgs);
                    }
                }
                EditorPythonRunnerRequestBus::Broadcast(ExecuteFilenamesWithArgs);

```